pkg/storage/logfs/locallog.go

```diff
@@ -258,6 +259,88 @@ func (l *localLog) QueryRecords(ctx context.Context, request storage.QueryRecord
 	return res, len(res) >= limit || totalSize >= l.cfg.MaxBunchSize, nil
 }
 
+// CountRecords count total number for records in the log and number of records after (before) specified record ID
+// Returned values are (total, count, error)
+func (l *localLog) CountRecords(ctx context.Context, request storage.QueryRecordsRequest) (uint64, uint64, error) {
```

Comment:
Comment: this looks good for the common case (and so far), but can be a little bit optimized in case of no filter provided. This case we don't need all the chunks be downloaded, but just calculate the number of records through metadata we got from Logs.

Can be done later...